pkg/security/x509source.go

```diff
@@ -476,18 +485,29 @@ func renewalTime(notBefore, notAfter time.Time) time.Time {
 	return notBefore.Add(notAfter.Sub(notBefore) * 7 / 10)
 }
 
+func contains(val string, list []string) bool {
+	for _, i := range list {
+		if i == val {
+			return true
+		}
+	}
+	return false
+}
+
 // isControlPlaneService returns true if the app ID corresponds to a Dapr
 // control plane service.
-func isControlPlaneService(id string) bool {
+func isControlPlaneService(id string, additionalCPServices []string) bool {
 	switch id {
 	case "dapr-operator",
 		"dapr-placement",
 		"dapr-injector",
 		"dapr-sentry":
 		return true
-	default:
-		return false
 	}
+	if additionalCPServices != nil {
+		return contains(id, additionalCPServices)
+	}
```

Comment:
Nit: the stdlib has a built-in method for this:


```suggestion
	if len(additionalCPServices) != 0 {
		return slices.Contains(additionalCPServices, id)
	}
```